backend/app/core/config.py

```diff
@@ -13,10 +14,32 @@ class ServerSessionsSettings(BaseSettings):
     expires_in: int
 
 
+class MatrixBotSettings(BaseSettings):
+    homeserver: str
+    access_token: str
+
+    # This checks if the user_id and homeserver match the rules set by the matrix spec.
+    # This returns a dictionary instead of a MatrixBotSettings object
+    @root_validator(pre=True)
+    def validate(cls, values):
+        homeserver = values.get("homeserver")
+
+        if re.search(r"https?://", homeserver) is None:
```

Comment:
It is possible to run against a http only homeserver, mainly only for local testing etc.